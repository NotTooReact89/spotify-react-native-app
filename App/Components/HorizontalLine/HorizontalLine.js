// @flow

import React from 'react'
import { View } from 'react-native'
import Styles from './HorizontalLineStyles'

export type HorizontalLineProps = {
  light?: boolean,
  inverse?: boolean,
  style?: any, // $FlowFixMe
}

const HorizontalLine = ({ light, inverse, style }: HorizontalLineProps) => {
  let lineStyle = null
  if (light && inverse) {
    lineStyle = Styles.lightInverse
  } else if (light) {
    lineStyle = Styles.light
  } else if (inverse) {
    lineStyle = Styles.inverse
  } else {
    lineStyle = Styles.default
  }

  return <View style={[Styles.horizontalLine, lineStyle, style]} />
}

HorizontalLine.defaultProps = {
  light: false,
  inverse: false,
  style: {},
}

export default HorizontalLine
