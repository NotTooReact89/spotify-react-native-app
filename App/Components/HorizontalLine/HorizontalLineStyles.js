// @flow

import { StyleSheet } from 'react-native'
import { Spaces, Colors } from '../../Styles'

export default StyleSheet.create({
  horizontalLine: {
    width: '100%',
    borderBottomWidth: 1,
    marginVertical: Spaces.sm,
  },
  light: {
    borderBottomColor: Colors.base.blackLight,
  },
  lightInverse: {
    borderBottomColor: Colors.base.whiteLight,
  },
  default: {
    borderBottomColor: Colors.base.text,
  },
  inverse: {
    borderBottomColor: Colors.base.textInverse,
  },
})
