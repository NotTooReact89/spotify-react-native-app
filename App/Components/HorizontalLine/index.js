// @flow

export { default } from './HorizontalLine'
