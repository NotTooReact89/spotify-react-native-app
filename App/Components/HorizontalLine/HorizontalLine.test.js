// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import HorizontalLine from './HorizontalLine'

jest.mock('Dimensions')

it('should render a light HorizontalLine correctly', () => {
  const tree = renderer.create(<HorizontalLine />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render a light inverse HorizontalLine correctly', () => {
  const tree = renderer.create(<HorizontalLine inverse />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render a strong HorizontalLine correctly', () => {
  const tree = renderer.create(<HorizontalLine strong />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render a strong inverse HorizontalLine correctly', () => {
  const tree = renderer.create(<HorizontalLine strong inverse />).toJSON()
  expect(tree).toMatchSnapshot()
})
