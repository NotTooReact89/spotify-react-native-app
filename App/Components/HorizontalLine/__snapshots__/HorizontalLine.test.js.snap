// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a light HorizontalLine correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "marginVertical": 6,
        "width": "100%",
      },
      Object {
        "borderBottomColor": "#000000",
      },
      Object {},
    ]
  }
/>
`;

exports[`should render a light inverse HorizontalLine correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "marginVertical": 6,
        "width": "100%",
      },
      Object {
        "borderBottomColor": "#ffffff",
      },
      Object {},
    ]
  }
/>
`;

exports[`should render a strong HorizontalLine correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "marginVertical": 6,
        "width": "100%",
      },
      Object {
        "borderBottomColor": "#000000",
      },
      Object {},
    ]
  }
/>
`;

exports[`should render a strong inverse HorizontalLine correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "marginVertical": 6,
        "width": "100%",
      },
      Object {
        "borderBottomColor": "#ffffff",
      },
      Object {},
    ]
  }
/>
`;
